pg_shard.c

```diff
@@ -586,6 +607,25 @@ SelectFromMultipleShards(Query *query, List *queryShardList)
 
 
 /*
+ * TODO: Add documentation
+ */
+static Query *
+BuildLocalQuery(Query *query, List *localRestrictList)
```

Comment:
Function naming: Do we expect this function to do anything apart from updating the where-clause list? If so, should we use a more specific name here? Something like UpdateRestrictClauses()?